sqlmesh/core/dialect.py

```diff
@@ -775,10 +781,24 @@ def jinja_statement(statement: str) -> JinjaStatement:
     return JinjaStatement(this=exp.Literal.string(statement.strip()))
 
 
+def _is_virtual_statement_begin(tokens: t.List[Token], pos: int) -> bool:
+    return _is_command_statement(ON_VIRTUAL_UPDATE_BEGIN, tokens, pos)
+
+
+def _is_virtual_statement_end(tokens: t.List[Token], pos: int) -> bool:
+    return _is_command_statement(ON_VIRTUAL_UPDATE_END, tokens, pos)
+
+
+def virtual_statement(statement: t.List[exp.Expression]) -> VirtualUpdateStatement:
+    return VirtualUpdateStatement(this=statement)
```

Comment:
If you're using `VirtualUpdateStatement` as a wrapper around a list of `Expression`s, then you should override `arg_types` so it only contains `"expressions": True`. Using `this` to store a list is misleading, because it's commonly used to store a single `Expression` instance.

Reply:
Right thank you revising this